lib/spack/spack/hooks/sbang.py

```diff
@@ -147,11 +151,39 @@ def install_sbang():
             spack.paths.sbang_script, sbang_path):
         return
 
-    # make $install_tree/bin and copy in a new version of sbang if needed
+    # remove outdated sbang if it exists
+    if os.path.exists(sbang_path):
+        os.remove(sbang_path)
+
+    # make $install_tree/bin
     sbang_bin_dir = os.path.dirname(sbang_path)
     fs.mkdirp(sbang_bin_dir)
-    fs.install(spack.paths.sbang_script, sbang_path)
-    fs.set_install_permissions(sbang_bin_dir)
+
```

Comment:
IIRC group members must still have at least execute permissions on the `sbang_bin_dir` in order to traverse/access contents of the directory.  I recommend retaining the permissions call that was at line 154, i.e.:

```suggestion
    fs.set_install_permissions(sbang_bin_dir)
    
```

Reply:
This would set the directory permissions to 0o755, which doesn't allow other members of your the group to write files (like `sbang`). That's why I got rid of it.

Reply:
I thought line 180 sets it.  In my testing, group and world ha no access to the directory whatsoever even when I set the package permissions to give write access.

Reply:
@tldahlgren I just updated in 0e3f4f0 - is that what you had in mind?